File: backend/app/schema/health.py
"""健康检查 DTO —— 拆出 main.py 到独立 schema 文件,避免 bootstrap 文件膨胀。"""

from __future__ import annotations

from pydantic import BaseModel


class HealthResponse(BaseModel):
    """存活探针响应 —— 表示 FastAPI 进程存活。"""

    status: str = "ok"


class ReadyResponse(BaseModel):
    """就绪探针响应 —— 每个依赖并行探测。"""

    status: str
    deps: dict[str, str]

File: frontend/src/types/health.ts
/**
 * 健康检查类型定义 —— 与后端 schema/health.py 对齐
 *
 * 后端端点:
 * - GET /health: 存活探针(永远 200)
 * - GET /ready: 就绪探针(可能返回 degraded)
 */

/** 存活探针响应 */
export interface HealthResponse {
  readonly status: "ok";
}

/** 就绪探针响应 */
export interface ReadyResponse {
  readonly status: "ok" | "degraded";
  readonly deps: Readonly<Record<string, "ok" | "fail">>;
}
